### PR TITLE
syncer: sync region with statistics

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
tidb-dashboard needs followers to know about the statistic of regions.

### What is changed and how it works?
This PR support syncs regions with their statistics.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -60,3 +60,5 @@ require (
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0
 	gopkg.in/yaml.v2 v2.2.8 // indirect
 )
+
+replace github.com/pingcap/kvproto => github.com/rleungx/kvproto v0.0.0-20200305052806-a32ceefc091e
```
Comment (nolouch) [inline go.mod]: Where is your PR about this?

Comment (rleungx) [reply go.mod]: https://github.com/rleungx/kvproto/tree/sync-statistic

File: server/cluster/cluster.go [hunk 2]
```diff
@@ -503,6 +503,8 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 				zap.Error(err))
 		}
 		regionEventCounter.WithLabelValues("update_kv").Inc()
+	}
+	if saveKV || saveCache {
```
Comment (nolouch) [inline server/cluster/cluster.go]: Maybe only statistic changed?

**Pull request merged**